packages/job-components/src/operations/core/slicer-core.ts

```diff
@@ -58,6 +58,9 @@ export default abstract class SlicerCore<T = OpConfig>
      * @param recoveryData is the data to recover from
      */
     async initialize(recoveryData: SlicerRecoveryData[]): Promise<void> {
+        if (recoveryData?.length > 0 && !this.isRecoverable()) {
+            throw new Error('cannot provide recovery data to a slicer that is not recoverable. Please create the isRecoverable method and have it return true if recovery is desired');
```

Comment:
Errors should start with a capital letter, (especially when the second sentence that starts with a capital letter). While this error message would make more sense within the test harness, it may be confusing when running the job since the person using the reader isn't given it "recovery data", it is just recovering the slicer.

```suggestion
            throw new Error('Slicer is not recoverable, please create the isRecoverable method and return true to enable recovery');
```

We should throw this error in the constructor too so it will fail before it actually starts recovering the previous slices. This can be achieved by using the `recovered_execution` boolean on the [execution config](https://github.com/terascope/teraslice/blob/d09841be3ac4d2dd33ce02a3c447e1a7a39f634d/packages/job-components/src/interfaces/jobs.ts#L105).

Reply:
yeah, moving it to the constructor is a better idea